api/utils/helpers.py

```diff
@@ -1,9 +1,91 @@
-from flask import json, Response, request
+import json
+import os
+import logging
+from types import GeneratorType
+
 from sqlalchemy_utils import sort_query
 
 
+def responsify(body, code, headers=None):
+    default_headers = {"Access-Control-Allow-Origin": "*", "Access-Control-Allow-Credentials": True}
+    default_headers.update(headers) if headers else None
+    results_obj = {}
+    if isinstance(body, dict):
+        results_obj = body
+    elif isinstance(body, list) or isinstance(body, str):
+        results_obj = {"data": body}
+    elif isinstance(body, GeneratorType):
+        results_obj = {"data": [item for item in body]}  # Unpack generator
+    return {
+        "statusCode": code,
+        "isBase64Encoded": False,
+        "headers": default_headers,
+        "body": json.dumps(results_obj)
+    }
+
+
+class Abort(Exception):
+    def __init__(self, code, reason):
+        msg = f"ERROR:{code} MSG: {reason}"
+        super(Abort, self).__init__(msg)
+
+
+class Serverless:
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    args = {}
+
+    class Args(object):
+        def __init__(self, parameters={}):
+            self.params = parameters
+
+        def get(self, key, default, type=str):
+            try:
+                value = self.params.get(key)
+                if value:
+                    return type(value)
+                else:
+                    return default
+            except TypeError:
+                raise Abort(400, f"{type} is not a valid Type")
+            except ValueError:
+                return Abort(400, f"Invalid literal {key} for type {type}")
+
+    @classmethod
+    def lambda_handler(cls, event):
+        cls.logger.info('## ENVIRONMENT VARIABLES')
+        cls.logger.info(os.environ)
+        cls.logger.info('## EVENT')
+        cls.logger.info(event)
+
+    @classmethod
+    def route(cls, func):
+        def wrapper(event: dict, _):
+            cls.lambda_handler(event)
+            stringParams: dict = event.get('queryStringParameters') or {}
+            pathParams: dict = event.get('pathParameters') or {}
+            cls.args = cls.Args(stringParams)
+
+            cls.logger.info(f"Arguments {stringParams} | {pathParams}")
+
+            try:
+                body, *context = func(pathParams) if pathParams else func()
+                code, headers = context if len(context) > 1 else (context[0], None)
+                res = responsify(body, code, headers)
+                cls.logger.info(res)
+                return res
+            except Exception as e:
+                return responsify(f"An unknown error occurred {e}", 400)
```

Comment:
Shouldn't this be a 500 error code?

Reply:
debatable
400 -> Client error
500 -> Server error

I figured it was more likely that the client passed in a bad argument than the server failing to do something